test/schema/examples.test.ts

```diff
@@ -0,0 +1,20 @@
+import * as reflect from 'jsii-reflect';
+import * as jsonschema from 'jsonschema';
+import { Schema } from 'jsonschema';
+import { readTemplate } from '../../src';
+import { renderFullSchema } from '../../src/cdk-schema';
+import { testExamples, Testing } from '../util';
+
+let typeSystem: reflect.TypeSystem;
+let schema: Schema;
+beforeAll(async () => {
+  typeSystem = await Testing.typeSystem;
+  schema = renderFullSchema(typeSystem);
+});
+
+testExamples(async (example) => {
```

Comment:
We currently have:

* A set of templates that we test for syntatically valid syntax
* A completely separate set of tests that check that some template syntax semantically parses to something else

It concerns me that we have no guarantee that we test all possible forms completely. Who is to say that the set of syntax checks overlaps completely with the set of semantic checks? 

Can we do a JSON check in every parsed template in the test of unit tests, or something similar, to make sure that we 100% know that our coverage of schema validation overlaps with the coverage of features?